### PR TITLE
MAKE-929: extend depot interface

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

**Comment by tychoish** [comment]
evergreen retry

File: credentials.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+		return nil, errors.Wrap(err, "error opening credentials file")
+	}
+	defer file.Close()
+
+	contents, err := ioutil.ReadAll(file)
+	if err != nil {
+		return nil, errors.Wrap(err, "error reading credentials file")
+	}
+
+	// TODO
```
Comment (Kimchelly) [inline credentials.go]: Do we still need this?

File: superset.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	}
+
+	pemCrt, err := crt.Export()
+	if err != nil {
+		return nil, errors.Wrap(err, "problem exporting certificate")
+	}
+
+	creds, err := NewCredentials(pemCACrt, pemCrt, pemKey)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create RPC credentials")
```
Comment (Kimchelly) [inline superset.go]: Remove "RPC"

File: superset.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	}
+
+	key, err := dpt.Get(PrivKeyTag(name))
+	if err != nil {
+		return nil, errors.Wrap(err, "problem getting key")
+	}
+
+	creds, err := NewCredentials(caCrt, crt, key)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create RPC credentials")
```
Comment (Kimchelly) [inline superset.go]: Remove "RPC".

**Pull request merged**